KalkulierbaR/kalkulierbar
# Sequent: Fix exRight and allLeft

Fixes #104 

Changes:
- Backend:
  - AllLeft, AllRight, ExLeft, and ExRight no longer take parameter `varAssign` of type `Map<String, String>` but `instTerm` of type `String`. This is a simplification, as the map contained only one element anyway and that must be an assignment for the bound variable by the respective quantifier.
  - `instTerm` is now allowed to be _any_ String that can be parsed to a term---i.e., not just a constant---for AllLeft and ExRight
  - Add `Signature` class representing the signature of a formula and `SignatureExtractor`
  - Replace `IdentifierCollector` with `SignatureExtractor` (=> we only need to search the names of, e.g., functions)
  - Check arity of functions/relations during parsing
    - Prevent using the same identifier for a function and constant / a relation and bound var
  - Check arity when instantiating a term
  - Implement automatic constant finding for AllRight/ExLeft
  - Tests for the new functionality
- Frontend:
  - Updated many dependencies
  - Bumped node version to 18
  - No longer rely on `preact-cli` as it is outdated
  - Now build with [vite](https://vitejs.dev/) 
  - Transformed the scss (style) files to scss modules (a new standard)
  - Switch from TSLint to ESLint, as TSLint is no longer suppoerted
  - Change frontend according to backend API changes

TODO:
- [x] Arity checks for other calculi
- [x] Add automatic button to frontend for AllRight/ExLeft
- [x] Tests for Signature stuff
